tests/Resources/fs.addontest.js

```diff
@@ -0,0 +1,1493 @@
+				should.fail(true, false, 'expected fs.rmdirSync to throw Error when deleting non-empty directory');
+			} catch (error) {
+				error.name.should.eql('Error');
+				error.message.should.eql(`ENOTEMPTY: directory not empty, rmdir '${dirName}'`);
+				error.errno.should.eql(-66);
+				error.syscall.should.eql('rmdir');
+				error.code.should.eql('ENOTEMPTY');
+				error.path.should.eql(dirName);
+			}
+			should(fs.existsSync(dirName)).eql(true);
+		});
+
+		it('throws trying to remove file instead of directory', () => {
+			try {
+				fs.rmdirSync(thisFilePath);
+				should.fail(true, false, 'expected fs.rmdirSync to throw Error when deleting file');
+			} catch (error) {
+				error.name.should.eql('Error');
+				error.message.should.eql(`ENOTDIR: not a directory, rmdir '${thisFilePath}'`);
+				error.errno.should.eql(-20);
+				error.syscall.should.eql('rmdir');
+				error.code.should.eql('ENOTDIR');
+				error.path.should.eql(thisFilePath);
+			}
+		});
+
+		it('throws trying to remove non-existent directory', () => {
+			const dirName = '/made/up/path';
+			should(fs.existsSync(dirName)).eql(false);
+
+			try {
+				fs.rmdirSync(dirName);
+				should.fail(true, false, 'expected fs.rmdirSync to throw Error when deleting dir that does not exist');
+			} catch (error) {
+				error.name.should.eql('Error');
+				error.message.should.eql(`ENOENT: no such file or directory, rmdir '${dirName}'`);
+				error.errno.should.eql(-2);
+				error.syscall.should.eql('rmdir');
+				error.code.should.eql('ENOENT');
+				error.path.should.eql(dirName);
+			}
+		});
+	});
+
+	describe('#stat()', () => {
+		it('is a function', () => {
+			should(fs.stat).be.a.Function;
+		});
+
+		it('returns stats for this file', finished => {
+			fs.stat(thisFilePath, (err, stats) => {
+				try {
+					should(stats).be.ok;
+					should(stats).be.an.Object;
+
+					// TODO: Verify some of the values?
+					finished();
+				} catch (e) {
+					finished(e);
+				}
+			});
+		});
+	});
+
+	describe('#statSync()', () => {
+		it('is a function', () => {
+			should(fs.statSync).be.a.Function;
+		});
+
+		it('returns stats for this file', () => {
+			const stats = fs.statSync(thisFilePath);
+			should(stats).be.ok;
+			should(stats).be.an.Object;
+
+			stats.size.should.be.above(0);
+			stats.blocks.should.be.above(0);
+
+			// check ctime and mtime versus Ti.Filesystem.File modifiedAt/createdAt?
+			stats.ctime.should.eql(thisFile.createdAt());
+			stats.mtime.should.eql(thisFile.modifiedAt());
+
+			stats.isFile().should.eql(true);
+			stats.isDirectory().should.eql(false);
+			// TODO Verify isSocket()/isCharacterDevice()/isBlockDevice()/isFIFO()/isSymbolicLink()?
+		});
+	});
+
+	describe('#truncate()', () => {
+		it('is a function', () => {
+			should(fs.truncate).be.a.Function;
+		});
+
+		it('truncates to 0 bytes by default', finished => {
+			const dest = Ti.Filesystem.tempDirectory + `truncate_${Date.now()}.js`;
+			fs.copyFileSync(thisFilePath, dest);
```

Comment:
<!--
  0 failure: 
  1 warning:  tests/Resources/f...
  
  
  DangerID: danger-id-Danger;
  File: tests/Resources/fs.addontest.js;
  Line: 997;
-->

- :warning: tests/Resources/fs.addontest.js line 997 – The 'fs.copyFileSync' is not supported until Node.js 8.5.0. The configured version range is '>=8'. (node/no-unsupported-features/node-builtins)